test/test_arraycontext.py

```diff
@@ -120,6 +120,14 @@ def __init__(self, actx, data):
 
     __array_priority__ = 10
 
+    def __bool__(self):
+        if len(self) == 1 and len(self.data[0]) == 1:
+            return bool(self.data[0])
```

Comment:
We should probably check the truth value of the to-numpy-ed scalar.

Reply:
You mean add a test that actually exercises this?

Reply:
> do self.array_context.to_numpy(self.data[0]) != 0?

I meant this. (sorry for the ambiguity)

Reply:
Resolving since you already implemented it.